xbmc/addons/AddonDatabase.cpp

```diff
@@ -305,16 +305,16 @@ void CAddonDatabase::SyncInstalled(const std::set<std::string>& ids,
     for (const auto& id : system)
     {
       m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID='%s'", id.c_str()));
-      // Set origin *only* for addons that do not have one yet as it may have been changed by an update.
-      m_pDS->exec(PrepareSQL("UPDATE installed SET origin='%s' WHERE addonID='%s' AND origin=''",
-          ORIGIN_SYSTEM, id.c_str()));
+      // Make sure system addons always have ORIGIN_SYSTEM origin
+      m_pDS->exec(PrepareSQL("UPDATE installed SET origin='%s' WHERE addonID='%s'", ORIGIN_SYSTEM,
+                             id.c_str()));
```

Comment:
It might be better to bump the DB version and only run this on that version change rather than on every sync as it should only be required a single time.

Reply:
guess this would have an impact if system addons get added to `addon-manifest.xml` after the db bump, as their origin wouldn't update. latest addition was 4 months ago with `<addon>kodi.binary.global.tools</addon>`. on the other hand what happens if a system addon gets removed/deprecated? (maybe unlikely but not impossible)

Reply:
I had a or ready to remove origin check and replace issytemaddon. Which is made at.boot time from the manifest. Then origin for system addons is irrelevant

Reply:
If that was the case then origin for installed addons would essentially be where the last install/update came from. Is that useful in anyway or is being consistent better?

I think that because system addons are installed with kodi the removal case is covered (I think).

Reply:
yup, their origin would get updated in db to whatever repo last updated them.
So would allow showing the repo that they got their last update from instead of "system add-on" (if wanted).
Before that they would just have empty origin until they first update.